framework/src/base/ComputeResidualThread.C

```diff
@@ -91,6 +91,24 @@ ComputeResidualThread::subdomainChanged()
     }
   }
 
+  // Interface DG Kernel dependencies
+  for (std::set<unsigned int>::const_iterator id_it = subdomain_boundary_ids.begin();
+      id_it != subdomain_boundary_ids.end();
+      ++id_it)
+  {
+    std::vector<DGKernel *> interface_dgks;
+    _sys.getDGKernelWarehouse(_tid).activeInterfaceDG(*id_it, interface_dgks);
```

Comment:
Typically the warehouse interface returns a reference to a container of objects, it is not filling up references.
Is that still correct for our warehouse refactoring @aeslaughter ?